src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/CreateActionPlanTest.kt

```diff
@@ -181,98 +184,100 @@ class CreateActionPlanTest : IntegrationTestBase() {
       .isCreated()
 
     // Then
-    val actionPlan = actionPlanRepository.findByPrisonNumber(prisonNumber)
+    val actionPlan = getActionPlan(prisonNumber)
     assertThat(actionPlan)
       .isForPrisonNumber(prisonNumber)
-      .hasNumberOfGoals(1)
-      .wasCreatedBy(dpsUsername)
-    val goal = actionPlan!!.goals!![0]
-    assertThat(goal)
-      .hasTitle(createGoalRequest.title)
-      .hasNumberOfSteps(createGoalRequest.steps.size)
-      .wasCreatedAtPrison(createGoalRequest.prisonId)
-      .wasCreatedBy(dpsUsername)
-      .hasCreatedByDisplayName(displayName)
-      .wasUpdatedBy(dpsUsername)
-      .hasUpdatedByDisplayName(displayName)
-      .wasUpdatedAtPrison(createGoalRequest.prisonId)
-    val step = goal.steps!![0]
-    assertThat(step)
-      .hasTitle(createStepRequest.title)
-      .hasStatus(StepStatus.NOT_STARTED)
-      .wasCreatedBy(dpsUsername)
-
-    val notes = noteRepository.findAllByEntityReferenceAndEntityTypeAndNoteType(actionPlan.goals!![0].reference!!, EntityType.GOAL, NoteType.GOAL)
-    Assertions.assertThat(notes.size).isGreaterThan(0)
-    Assertions.assertThat(notes[0].content).isEqualTo("Chris would like to improve his listening skills, not just his verbal communication")
+      .hasNumberOfGoals(2)
+      // Because the goals are returned in date created order, the first goal in the response will be Goal 2, because it was the most recently created.
+      .goal(1) {
+        it.hasTitle("Goal 2")
+          .hasNumberOfSteps(1)
+          .step(1) {
+            it.hasTitle("Step 1 of Goal 2")
+              .hasStatus(StepStatus.NOT_STARTED)
+          }
+          .hasGoalNote("Goal2 notes")
+      }
+      // The 2nd goal in the response data is the first goal to have been created
+      .goal(2) {
+        it.hasTitle("Goal 1")
+          .hasNumberOfSteps(1)
+          .step(1) {
+            it.hasTitle("Step 1 of Goal 1")
+              .hasStatus(StepStatus.NOT_STARTED)
+          }
+          .hasNoNotes()
+      }
+      .allGoals {
+        it.wasCreatedAtPrison(createGoalRequest1.prisonId)
+          .wasCreatedBy(dpsUsername)
+          .hasCreatedByDisplayName(displayName)
+          .wasUpdatedBy(dpsUsername)
+          .hasUpdatedByDisplayName(displayName)
+          .wasUpdatedAtPrison(createGoalRequest1.prisonId)
+      }
   }
 
   @Test
-  @Transactional
-  fun `should create a new action plan with multiple goals given prisoner does not have an action plan`() {
+  fun `should create a new action plan and not create a review schedule given the prisoner does not have an Induction created before the Action Plan`() {
```

Comment:
New tests to create an action plan before induction and assert no review schedule created; and a test to create the action plan after the induction and to assert the review schedule is created.